### PR TITLE
[Routing] Fail properly when a route parameter name cannot be used as a PCRE subpattern name

### PR DESCRIPTION
**Pull request by fancyweb**
| Q             | A
| ------------- | ---
| Branch?       | 2.7
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | https://github.com/symfony/symfony-docs/pull/7139

This is a follow up PR to https://github.com/symfony/symfony/pull/20327.

I also decided to improve the current `is_numeric()` check that is done by a truer one. The current limitation is that a PCRE subpattern name must not start with a digit which is different.

### CODE REVIEW

File: src/Symfony/Component/Routing/RouteCompiler.php [hunk 1]
```diff
@@ -29,11 +29,19 @@ class RouteCompiler implements RouteCompilerInterface
     const SEPARATORS = '/,;.:-_~+*=@|';
 
     /**
+     * The maximum supported length of a PCRE subpattern name
+     * http://pcre.org/current/doc/html/pcre2pattern.html#SEC16.
+     *
+     * @var int
+     */
+    const VARIABLE_MAXIMUM_LENGTH = 32;
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Routing/RouteCompiler.php]: please add `@internal` also so we don't have to apply the bc policy on this constant

**Comment by nicolas-grekas** [comment]
:+1:

**Comment by stof** [comment]
👍

**Comment by nicolas-grekas** [comment]
Thank you @fancyweb.

**Pull request closed without merging**